DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala

```diff
@@ -563,7 +565,40 @@ object CommonProcessorFactory {
         batchLog.warn(s"End batch ${batchTime}, output partition time:${outputPartitionTime}, namespace:${namespace}")
 
         metrics ++ result
-      } // end of processPaths
+      }, // end of processPaths
+      /*
+      process a batch of ConsumerRecords from kafka
+     */
+      processConsumerRecord = (rdd: RDD[ConsumerRecord[String,String]], batchTime: Timestamp, batchInterval: Duration, outputPartitionTime: Timestamp) =>{
+        processDataset(rdd
+          .map(d=>{
+            val value = d.value()
+            if(value==null) throw new EngineException(s"null bytes from ConsumerRecord")
```

Comment:
should we just eat the null value and send out a warning instead?

Reply:
We currently throw if input data is null for eventhub , this change is following the same behavior for kafka as well.